state/subnets.go

```diff
@@ -327,55 +334,43 @@ func (st *State) AddSubnet(args network.SubnetInfo) (subnet *Subnet, err error)
 				return nil, errors.Trace(err)
 			}
 		}
+		var ops []txn.Op
+		var subDoc subnetDoc
+		subDoc, ops, err = st.addSubnetOps(strconv.Itoa(seq), args)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		subnet = &Subnet{st: st, doc: subDoc}
+		ops = append(ops, assertModelActiveOp(st.ModelUUID()))
 		return ops, nil
 	}
 	err = st.db().Run(buildTxn)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	if err := subnet.setSpace(nil); err != nil {
```

Comment:
Why do we set this to nil? Does that mean "use the default space"?

Reply:
oh wait, this is a collection?

Reply:
The arg is nil because there isn't a mongo collection available pass.  Some places that call setSpace() already have the subnet collection available, others do not.  Therefore setSpace() will only get the subnet collection if it's not already available.